phoenicis-tools/src/main/java/org/phoenicis/tools/processes/PhoenicisProcess.java

```diff
@@ -0,0 +1,53 @@
+package org.phoenicis.tools.processes;
+
+import org.apache.commons.io.IOUtils;
+
+import java.io.IOException;
+import java.nio.charset.Charset;
+
+/**
+ * A wrapper class around the Java {@link Process} class allowing for a save access to the underlying process from the scripts
+ */
+public class PhoenicisProcess {
```

Comment:
We could think about moving this whole class to JS. This would mean that `ProcessUtils#startProcess` would return a Java `Process` object, which is then wrapped inside a JSObject in a proxy script.

Reply:
In general, is better to have our own class because it makes us sure that there are no unsafe things that are exposed. So I like your implementation

Reply:
My personal dilemma is the assessment between safety (like you mentioned) and ease of use.

If we implement this on the Java side we have full control over how things are processed and whether something should be prevented/disallowed for certain input.

If we implement this on the JavaScript side, we lose a lot of control because scripts can change or reload other scripts without Phoenicis being aware of this. On the other hand it is more more clear what functionality is available from inside the Scripts and how it needs to be accessed syntactically.

Reply:
In this case, I would vote for safety. Actually, this might not even be safe enough because it still allows you to spawn random processes (not that I would know any alternative to that in our special case).
One other point to consider: returning a Java object ties the scripts to Java whereas a Bean could be implemented in any language.

Reply:
Done

Reply:
+1